src/machines.rs

```diff
@@ -114,3 +128,40 @@ pub async fn update_machines_stats(
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use zbus::zvariant::OwnedObjectPath;
+
+    #[test]
+    fn test_filter_machines() {
+        let machines = vec![
+            crate::dbus::zbus_machines::ListedMachine {
+                name: "foo".to_string(),
+                class: "container".to_string(),
+                service: "".to_string(),
+                path: OwnedObjectPath::try_from("/sample/object").unwrap(),
+            },
+            crate::dbus::zbus_machines::ListedMachine {
+                name: "bar".to_string(),
+                class: "container".to_string(),
+                service: "".to_string(),
+                path: OwnedObjectPath::try_from("/sample/object").unwrap(),
+            },
+            crate::dbus::zbus_machines::ListedMachine {
+                name: "baz".to_string(),
+                class: "container".to_string(),
+                service: "".to_string(),
+                path: OwnedObjectPath::try_from("/sample/object").unwrap(),
+            },
+        ];
+        let allowlist = vec!["foo".to_string(), "baz".to_string()];
+        let blocklist = vec!["bar".to_string()];
```

Comment:
What's the behavior if someone puts bar in both lists? I believe it will still be blocked lookin at the filter function. Should we test that or throw an error if someone does put same name in both lists?

Reply:
I follow the logic from `units` file - https://github.com/cooperlees/monitord/blob/main/src/units.rs#L234 .

My intuitive approach was to make allowlist and blocklist are mutually exclusive - we either define a list of hosts to monitor or block certain entries and be happy with the remaining set. After inspecting the code though I decided to keep it consistent with existing allow/block list logic, simply to avoid confusion.

Reply:
If it's identical then we're good. I wonder if we should combine the code to select then so it will always be the same (another PR tho)